pallets/utxo/src/lib.rs

```diff
@@ -562,6 +677,58 @@ pub mod pallet {
         Ok(().into())
     }
 
+    pub fn tokens_create<T: Config>(
+        caller: &T::AccountId,
+        public: H256,
+        input_for_fee: TransactionInput,
+        token_name: String,
+        token_ticker: String,
+        supply: Value,
+    ) -> DispatchResultWithPostInfo {
+        ensure!(!supply.is_zero(), Error::<T>::MinBalanceZero);
+
+        // Take a free TokenID
+        let token_id =
+            <TokensHigherID<T>>::get().checked_add(1).ok_or("All tokens IDs has taken")?;
+        sp_runtime::print("TOKEN ID IS");
+        sp_runtime::print(&token_id);
+
+        // Input with MLT FEE
+        let fee = UtxoStore::<T>::get(input_for_fee.outpoint).unwrap().value;
+        ensure!(fee < 99, Error::<T>::Unapproved);
```

Comment:
So, about the first point, I read somewhere that `unwrap` in the Substrate does not make panics, correct me if I'm wrong. Nevertheless, we can change this place for a more comfortable view =)
```rust
let fee = UtxoStore::<T>::get(input_for_fee.outpoint).ok_or(Error::<T>::Unapproved)?.value;
```
The second point, I've double-checked it and you are right!

Reply:
Yes I think `.ok_or()` is preferable. Regarding `unwrap` not panicking, I did not find `Option` docs for substrate, they probably just use `Option` from Rust core. The substrate doc for `Result` suggest otherwise: https://substrate.dev/rustdocs/latest/sp_std/result/enum.Result.html#method.unwrap

Reply:
Okay I've just noticed something else. The 100 tokens here is in MLT which means this is the base unit (the sat of the MLT world), we really need a name for this base unit.

https://github.com/mintlayer/core/blob/36599cea0eafbac785a89ced09f0fdb3233b1042/node/src/chain_spec.rs#L150
You can see here in the genesis block that an MLT is defined as 1_000 * 100_000_000 base units so we don't have to worry about floating point stuff because that would give me nightmares.

This 100 should be 100 * 1_000 * 100_000_000

@anyxem we need to sort the UI so it understands this too. 1 unit is not 1 MLT just like 1 sat is not 1 btc.

Reply:
> we really need a name for this base unit.

Proposal - 100 * 1_000 * 100_000_000 Mint seeds == 100 MLT?

Reply:
The default weight for a token tx is fixed